src/client/power-pages/actions-hub/ActionsHubCommandHandlers.ts

```diff
@@ -259,3 +265,98 @@ export const uploadSite = async (siteTreeItem: SiteTreeItem) => {
+ * @returns Array of site information objects for sites found in the parent folder
+ */
+export function findOtherSites(knownSiteIds: Map<string, boolean>, fsModule = fs, yamlModule = yaml): IOtherSiteInfo[] {
+    // Get the workspace folders
+    const workspaceFolders = vscode.workspace.workspaceFolders;
+    if (!workspaceFolders || workspaceFolders.length === 0) {
+        return [];
+    }
+
+    const currentWorkspaceFolder = workspaceFolders[0].uri.fsPath;
+    const parentFolder = path.dirname(currentWorkspaceFolder);
+
+    try {
+        // Get directories in the parent folder
+        const items = fsModule.readdirSync(parentFolder, { withFileTypes: true });
+        const directories = items
+            .filter(item => item.isDirectory())
+            .map(item => path.join(parentFolder, item.name));
+
+        // Make sure we include the current workspace folder
+        if (!directories.includes(currentWorkspaceFolder)) {
+            directories.push(currentWorkspaceFolder);
+        }
+
+        const otherSites: IOtherSiteInfo[] = [];
+
+        // Check each directory for website.yml
+        for (const dir of directories) {
+            const websiteYamlPath = path.join(dir, WEBSITE_YML);
+
+            if (fsModule.existsSync(websiteYamlPath)) {
+                try {
+                    // Parse website.yml to get site details
+                    const yamlContent = fsModule.readFileSync(websiteYamlPath, UTF8_ENCODING);
+                    const websiteData = yamlModule.load(yamlContent) as WebsiteYaml;
+
+                    const websiteId = websiteData?.adx_websiteid;
+
+                    // Only include sites that aren't already in active or inactive sites
+                    if (websiteId && !knownSiteIds.has(websiteId.toLowerCase())) {
+                        otherSites.push({
+                            name: websiteData?.adx_name || "",
+                            websiteId: websiteId,
+                            folderPath: dir
+                        });
+                    }
+                } catch (error) {
+                    oneDSLoggerWrapper.getLogger().traceError(
+                        Constants.EventNames.OTHER_SITES_YAML_PARSE_FAILED,
+                        error as string,
+                        error as Error,
+                        { methodName: findOtherSites.name }
+                    );
+                }
+            }
+        }
+
+        return otherSites;
+    } catch (error) {
+        oneDSLoggerWrapper.getLogger().traceError(
+            Constants.EventNames.OTHER_SITES_FILESYSTEM_ERROR,
+            error as string,
+            error as Error,
+            { methodName: findOtherSites.name }
+        );
+        return [];
+    }
+}
+
+export function createKnownSiteIdsMap(
+    activeSites: IWebsiteDetails[] | undefined,
+    inactiveSites: IWebsiteDetails[] | undefined
+): Map<string, boolean> {
+    const knownSiteIds = new Map<string, boolean>();
```

Comment:
Do we need a `Map` here? Since we are just checking the existense I guess we can use `Set` instead. What do you think?

Reply:
Right, it will be more memory efficient.